.github/actions/templates/avm-validateModuleDeployment/action.yml

```diff
@@ -59,7 +59,54 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: Azure Login
+    # [Azure login] task(s)
+    # ------------------------------
+    - name: "Set OIDC exception"
+      id: set-oidc-exception
```

Comment:
Does this mean you'd always need both a service principal, and OIDC? If the case - is it worth even introducing? 
Maybe I'm wrong, but I'd argue that we only really get the full security benefits of OIDC if we using it all the way and not in addition as it then doesn't actually save us any credentials etc. 

In any case, I now get why this was such a painful feature to implement :o

Reply:
Personally, I need OIDC support, so I don't create SFI related issues in my testing sub.

Reply:
Fair, I guess that means you won't adopt any of the 'exception' resources in the list below any time soon :D 

I'm sure you looked it up @eriqua, but it would be a crime not to ask: Is there any way to extend the tokens validity explicitely? Say on the Azure-side?

Reply:
@AlexanderSehr I tried to summarize all findings and background work for this PR in [this comment](https://github.com/Azure/bicep-registry-modules/issues/2092#issuecomment-2221749638).

There, I mention an option to extend the token lifetime. The option is in preview and I'd foresee additional time to test it and to make sure it works for our contributors as well.

I hear you that keeping both authentication methods is not ideal. However, I believe this mid step is still worth integrating.
The majority of the modules' e2e tests require less than 1 hour to complete and would hence be covered by the OIDC authentication method. This means the majority of module owners and contributors may adopt OIDC only and remove AZURE_CREDENTIALS on their end. This imho is already an improvement looking at the broader picture.
In parallel, we can test the token lifetime extension or agree on adopting it once GA. Until then, we'll be able to familiarize with OIDC for more than the 80% of our modules, possibly dealing with other caveats in case they arise.

Reply:
Hey @eriqua,
thanks for elaborating. One thing caught my attention though: You write in your comment that an MSI has a way longer token validity. I recall sharing a video with you and the others some time ago where @matt-FFFFFF explains how to use an MSI for OIDC and it looked super simple. Given that + the fact that an MSI would anyways be preferable over a SPN, would that not be a better way to go? I think from the GH side nothing much changes and it's just a matter of creating the MSI & configuring it on the Azure-side, just as it would be for the SPN.
Curious to hear your thoughts :) And happy to give it a spin, of course.

Reply:
@AlexanderSehr I'd spend more time to reply to this comment than to update the background issue itself. Resolving to avoid confusion.

Reply:
@eriqua , thank you for all your work here.

I'm trying to understand how this gh-action's design is, and if there might be a way to use only OIDC, even for modules taking long time.

Everytime we use the step `uses: azure/login@v2` with OIDC, what we are doing is calling `az login --federatedToken xxxx` where the login@v2 action is getting the fedToken from gh issuer.  

When a bicep module is built and submited to ARM, we only need a JWT to submit the operation. I don't think we need a new JWT while we wait for the deployment, right?

So, can we update the CI workflow to call the step `azure/login@v2` before testing each bicep module?  For example, break the main workflow into stages where we first create the list of modules to tests, and then run:
- azure/login@v2
- testModule

This might allow running/testing all bicep modules in parallel, and let each test to get a token to submit the deployment request to ARM

Reply:
Hey @vhvb1989, please refer to latest implementation #2757, using OIDC with msi. It seems that even if using msi, auth-type: SERVICEPRINCIPAL is still required for GitHub hosted runners.
That seems to fix the issue with token expiration, and I could remove the workaround tested in this draft PR for longer deployments.
However, expiring tokens have still been noticed in a few occasions even when leveraging an msi, so testing will continue.

Reply:
Closing the comment in favor of #2757